consensus-types/types/genesis_test.go

```diff
@@ -34,10 +34,10 @@ import (
 
 func TestDefaultGenesis(t *testing.T) {
 	g := types.DefaultGenesis()
-	if g.ForkVersion != bytes.FromUint32(version.Deneb) {
+	if !g.ForkVersion.Equals(version.Deneb()) {
```

Comment:
so the equals here is the bytes.Equal, which is fine.
I wonder if we should dress it with a version.Equal?

Reply:
I explored keeping the comparable functions as utility functions in package `version`. In the end the readability of code felt better with putting these on the type itself with receiver.

Reply:
But it's more than we need and we suggest that we have a canonical way to order/compare bytes, which we don't have.
We do only have versions ordering

Reply:
done, enforced in sorting test